cameraview/src/main/java/com/otaliastudios/cameraview/CameraView.java

```diff
@@ -263,9 +273,42 @@ protected CameraPreview instantiatePreview(Context context, ViewGroup container)
     protected void onAttachedToWindow() {
         super.onAttachedToWindow();
         if (mCameraPreview == null) {
+
+            // check if we have any overlay view before instantiating preview
+            boolean foundOverlay = false;
+            for (int i = 0; i < getChildCount(); i++) {
+                View view = getChildAt(i);
+                if (view.getLayoutParams() instanceof OverlayLayoutParams &&
+                        ((OverlayLayoutParams) view.getLayoutParams()).isOverlay) {
+                    foundOverlay = true;
+                }
+            }
+            mHasOverlay = foundOverlay;
+
             // isHardwareAccelerated will return the real value only after we are
             // attached. That's why we instantiate the preview here.
             instantiatePreview();
+
+            mPreviewOverlayLayout = findViewById(R.id.preview_overlay_layout);
+
+            LinkedList<View> overlayViews = new LinkedList<>();
```

Comment:
I would remove all this part also. Instead you could:

- override addView() signature where you have the LayoutParams.
- if params is an instance of OverlayLayoutParams, call mOverlayLayout.addView(). If not, call super.addView().

This way we avoid moving views around & it's simpler